dispatcher_plugin_nb2workflow/exposer.py

```diff
@@ -93,28 +107,60 @@ def get_instr_conf(from_conf_file=None):
     
     return cfg_dict
 
-config_dict = get_instr_conf(conf_file)
-if 'ODA_ONTOLOGY_PATH' in os.environ:
-    ontology_path = os.environ.get('ODA_ONTOLOGY_PATH')
-else:
-    ontology_path = config_dict.get('ontology_path', 
-                                    'http://odahub.io/ontology/ontology.ttl')
-logger.info('Using ontology from %s', ontology_path)
+combined_instrument_dict = {}
+def build_combined_instrument_dict():
+    global combined_instrument_dict
+    combined_instrument_dict = copy(static_config_dict.get('instruments', {}))
+    combined_instrument_dict.update(get_config_dict_from_kg()['instruments'])
+
+build_combined_instrument_dict()
 
 def factory_factory(instr_name, restricted_access):
+    instrument_query = NB2WInstrumentQuery('instr_query', restricted_access)
     def instr_factory():
-        backend_options = NB2WDataDispatcher(instrument=instr_name).query_backend_options()
-        query_list, query_dict = NB2WProductQuery.query_list_and_dict_factory(backend_options, ontology_path)
+        backend_options = NB2WDataDispatcher(instrument=instr_name).backend_options
+        query_list, query_dict = NB2WProductQuery.query_list_and_dict_factory(backend_options, 
+                                                                              ontology_path)
         return Instrument(instr_name,
-                        src_query = NB2WSourceQuery.from_backend_options(backend_options, ontology_path),
-                        instrumet_query = NB2WInstrumentQuery('instr_query', restricted_access),
-                        data_serve_conf_file=conf_file,
+                        src_query = NB2WSourceQuery.from_backend_options(backend_options, 
+                                                                         ontology_path),
+                        instrumet_query = instrument_query,
+                        data_serve_conf_file=masked_conf_file,
                         product_queries_list=query_list,
                         query_dictionary=query_dict,
                         asynch=True, 
                         data_server_query_class=NB2WDataDispatcher,
                         )
+    instr_factory.instr_name = instr_name
+    instr_factory.instrument_query = instrument_query
     return instr_factory
```

Comment:
so now the factory itself has `instr_name` and `instrument_query`. Is it for performance?

Reply:
Yes this is performance optimisation. Also explained here https://github.com/oda-hub/dispatcher-app/pull/661#discussion_r1503880654